src/files-regular/add-pull-stream.js

```diff
@@ -44,35 +45,21 @@ module.exports = (common, options) => {
 
       const stream = ipfs.addPullStream()
 
-      pull(
-        pull.values(files),
-        stream,
-        pull.collect((err, filesAdded) => {
-          expect(err).to.not.exist()
+      const filesAdded = await pullToPromise.any(pull(pull.values(files), stream))
```

Comment:
IMO pull-streams pipelines should not be transformed in promises, you lose flexibility and gain 0 value. Pull-streams should either disappear in the future or be left to work as they were created.